### PR TITLE
ENH: Support for list of formats in pd.to_datetime() (pandas-dev#55226)

### PR DESCRIPTION
**Pull request by gusthavoMFS**
Trying to convert format list for each data list element.

The idea is to make the format argument more flexible to accept a list, in order to specify the format for each date in a list.

### CODE REVIEW

File: pandas/tests/tools/test_to_datetime.py [hunk 1]
```diff
@@ -1086,10 +1086,24 @@ def test_to_datetime_today_now_unicode_bytes(self, arg):
...
         # https://github.com/pandas-dev/pandas/issues/50359
         result = to_datetime(["2020-01-03 00:00:00Z", string], format=format, utc=True)
-        expected = DatetimeIndex(
+        expected = DatetimeIndex(d
             [expected_ds, getattr(Timestamp, attribute)()], dtype="datetime64[ns, UTC]"
         )
         assert (expected - result).max().total_seconds() < 1
+    
+    
+    #MUDEI AQUI
```
Comment (MarcoGorelli) [inline pandas/tests/tools/test_to_datetime.py:1095]: thanks for starting this - could you mark this as draft please and ping me when it's ready for review?

Comment (gusthavoMFS) [reply pandas/tests/tools/test_to_datetime.py]: Thanks for the encouragement, I've already put it as a draft

**Comment by gusthavoMFS** [comment]
Hello @MarcoGorelli we need a review for this code.
Thanks for attention

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
there's a tonne of irrelevant stuff in this PR, please remove it

**Comment by leogiuris** [comment]
Hello @MarcoGorelli , I reverted a commit with many graph images that had nothing to do with the issue. Can you review for us? Thank you for your patience.

File: pandas/core/tools/datetimes.py [hunk 2]
```diff
@@ -715,13 +718,14 @@ def to_datetime(
     ...
 
 
+#MUDANDO AQUI
```
Comment (MarcoGorelli) [inline pandas/core/tools/datetimes.py:721]: e.g. this

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
there's still several unrelated things

there's a lot of stuff to do, please make sure this is in good shape before asking for a review

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**